parley/src/tests/utils/env.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2024 the Parley Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::tests::utils::renderer::render_layout;
+use crate::{
+    FontContext, FontFamily, FontStack, Layout, LayoutContext, RangedBuilder, StyleProperty,
+};
+use peniko::Color;
+use std::path::{Path, PathBuf};
+
+// Creates a new instance of TestEnv and put current function name in constructor
+#[macro_export]
+macro_rules! testenv {
+    () => {{
+        // Get name of the current function
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        let name = type_name_of(f);
+        let name = &name[..name.len() - 3];
+        let name = &name[name.rfind(':').map(|x| x + 1).unwrap_or(0)..];
+
+        // Create test env
+        $crate::tests::utils::TestEnv::new(name)
+    }};
+}
+
+fn current_imgs_dir() -> PathBuf {
+    Path::new(env!("CARGO_MANIFEST_DIR"))
+        .join("tests")
+        .join("current")
+}
+
+fn snapshot_dir() -> PathBuf {
+    Path::new(env!("CARGO_MANIFEST_DIR"))
+        .join("tests")
+        .join("snapshots")
+}
+
+fn font_dir() -> PathBuf {
+    Path::new(env!("CARGO_MANIFEST_DIR"))
+        .join("tests")
+        .join("assets")
+        .join("roboto_fonts")
+}
+
+const DEFAULT_FONT_NAME: &str = "Roboto";
+
+pub(crate) struct TestEnv {
+    test_name: String,
+    check_counter: u32,
+    font_cx: FontContext,
+    layout_cx: LayoutContext<Color>,
+    foreground_color: Color,
+    background_color: Color,
+    tolerance: f32,
+    errors: Vec<(PathBuf, String)>,
+}
+
+use fontique::{Collection, CollectionOptions};
```

Comment:
Please leave all imports at the top of the file.

Reply:
Fixed